ddtrace/contrib/django/patch.py

```diff
@@ -93,15 +93,18 @@ def cursor(django, pin, func, instance, args, kwargs):
             "django.db.alias": alias,
         }
         pin = Pin(service, tags=tags, tracer=pin.tracer, app=prefix)
-        cursor = func(*args, **kwargs)
-        traced_cursor_cls = dbapi.TracedCursor
-        if (
-            Psycopg2TracedCursor is not None
-            and hasattr(cursor, "cursor")
-            and isinstance(cursor.cursor, psycopg_cursor_cls)
-        ):
-            traced_cursor_cls = Psycopg2TracedCursor
-        return traced_cursor_cls(cursor, pin, config.django)
+        cursor = func(*args, **kwargs)  # type: django.db.backends.utils.CursorWrapper
+
+        # Ensure the CursorWrapper.cursor is wrapped
+        if not isinstance(cursor.cursor, dbapi.TracedCursor):
+            traced_cursor_cls = dbapi.TracedCursor
+            if Psycopg2TracedCursor is not None and isinstance(cursor.cursor, psycopg_cursor_cls):
+                traced_cursor_cls = Psycopg2TracedCursor
+            setattr(cursor, "cursor", traced_cursor_cls(cursor.cursor, pin, config.django))
```

Comment:
_slightly_ concerned about this because of MRO issues (we really need a helper for that...).

We might end up adding an attribute to `cursor` that might exist on its parent

Reply:
not sure I follow here, `cursor` is a `CursorWrapper`, which we know will have `cursor` attribute ?